x-pack/plugins/osquery/server/routes/pack/update_pack_route.ts

```diff
@@ -349,8 +349,25 @@ export const updatePackRoute = (router: IRouter, osqueryContext: OsqueryAppConte
         );
       }
 
+      const { attributes } = updatedPackSO;
+
       return response.ok({
-        body: { data: { ...updatedPackSO.attributes, saved_object_id: updatedPackSO.id } },
+        body: {
```

Comment:
We're still allowing the response type to be implicitly defined by the `PackSavedObject` type here. We need something like https://github.com/elastic/kibana/pull/152111/files#diff-99c058e27bef314374389a57a0872a47d332713d84b40ab95999cd75413b8d1aR37 where a separate TS type is created and dedicated specifically to the HTTP API response. This way, if the internal Saved Object data structure changes in the future, the API response type will not be affected.

Reply:
Thanks @marshallmain, I specified data type, how does this look now?